sklearn/impute/_iterative.py

```diff
@@ -773,3 +773,6 @@ def fit(self, X, y=None):
         """
         self.fit_transform(X)
         return self
+
+    def _more_tags(self):
+        return {"allow_nan": is_scalar_nan(self.missing_values)}
```

Comment:
This seems unrelated to the original problem with `SimpleImputer`, right? If so, I would either remove this from this PR or add new tests for this and the k-NN imputer.

Reply:
This change is per @thomasjpfan 's request to move `_more_tags` from `_BaseImputer` down to the children (see https://github.com/scikit-learn/scikit-learn/pull/21114#discussion_r716034005). I don't fully understand why overriding doesn't work here though. Let me know if you have some insights.

Reply:
Sorry I was incorrect. I forgot that the MRO walk was reversed in `_get_tags`:

https://github.com/scikit-learn/scikit-learn/blob/9cb4e761f530e7a0708059b1312921753d056cbe/sklearn/base.py#L343-L345

**For this PR**, we only need to add a new `SimpleImputer._more_tags` and do not need to update the other imputers.

XREF: https://github.com/scikit-learn/scikit-learn/issues/20804

Reply:
Thanks for the answering. As suggested, I moved `_more_tags` back to the base class and added an override to `SimpleImputer`.